src/Servers/Kestrel/Core/src/Internal/Http/HttpParser.cs

```diff
@@ -305,14 +375,28 @@ private int ParseMultiSpanHeader(TRequestHandler handler, ref SequenceReader<byt
 
             if (headerSpan.Length < 5)
             {
-                // Less than min possible headerSpan is 5 bytes a:b\r\n
-                RejectRequestHeader(headerSpan);
+                if (!_enableLineFeedTerminator || headerSpan.Length < 4)
+                {
+                    // Less than min possible headerSpan is 4 bytes a:b\n or 5 bytes a:b\r\n
+                    RejectRequestHeader(headerSpan);
+                }
             }
 
             if (headerSpan[^2] != ByteCR)
             {
-                // Sequence needs to be CRLF not LF first.
-                RejectRequestHeader(headerSpan[..^1]);
+                if (!_enableLineFeedTerminator)
+                {
+                    // Sequence needs to be CRLF.
+                    RejectRequestHeader(headerSpan[..^1]);
+                }
+                else if (headerSpan[^2] != ByteLF)
+                {
+                    RejectRequestHeader(headerSpan[..^1]);
+                }
+                else
+                {
+                    return headerSpan.Length - 1;
```

Comment:
I understand the issue. Is there a more deterministic way to test this, maybe with a unit test and not a functional one. Because I can't hit this code path anymore.

Reply:
Added a test and fixed the implementation